### PR TITLE
DROOLS-1668 Fix some FindBugs problems

### PR DESCRIPTION
**Pull request by baldimir**
- Please check the fixes and if some of them are problematic, let me know.

### CODE REVIEW

**Comment by baldimir** [comment]
@etirelli @tarilabs can you please review this?

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/impl/FEELImpl.java [hunk 1]
```diff
@@ -72,7 +72,11 @@ public Object evaluate(String expression, Map<String, Object> inputVariables) {
             inputVariables.entrySet().stream().forEach( e -> ctx.addInputVariable( e.getKey(), e.getValue() ) );
         }
         CompiledExpression expr = compile( expression, ctx );
-        return evaluate( expr, inputVariables );
+        if ( inputVariables == null ) {
+            return evaluate( expr, Collections.emptyMap() );
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/impl/FEELImpl.java]: Minor, but I'd put `FEELImpl.EMPTY_INPUT` for consistency as per line 66, what do you think?

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/SubstringAfterFunction.java [hunk 2]
```diff
@@ -39,12 +39,9 @@ public SubstringAfterFunction() {
         int index = string.indexOf( match );
         if( index >= 0 ) {
             return FEELFnResult.ofResult( string.substring( index+match.length() ) );
-        } else if( index < 0 ) {
+        } else {
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/SubstringAfterFunction.java:42]: agree, the check for potential NPE also is done few lines above.

**Review by tarilabs — COMMENTED** [review]
Thank you!

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/impl/FEELImpl.java (hunk 1 above)
Comment (baldimir) [reply kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/impl/FEELImpl.java]: Fixed.

**Comment by tarilabs** [comment]
thanks again @baldimir

**Review by etirelli — APPROVED** [review]

**Pull request merged**